### PR TITLE
Rename function, add TODO

### LINKED ISSUE #1954: Refactor `types.NewPathFromString` to return `(Path, error)`
### What should be done?

`types.NewPathFromString` should return `(Path, error)` and not panic.

What errors are returned is not important in this issue. That's for #1744.

See #1873 for the original issue.

### Where?

https://github.com/FerretDB/FerretDB/blob/main/internal/types/path.go

### Definition of Done

- all handlers updated;
- unit tests added/updated;
- spot refactorings done;
- code comments updated (to at least explain what paths are invalid).

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #1954 (not really).

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/path.go [hunk 1]
```diff
@@ -69,26 +70,37 @@ type Path struct {
 	s []string
 }
 
-// NewPath returns Path from a strings slice.
-func NewPath(path ...string) Path {
+// NewStaticPath returns Path from a strings slice.
+//
+// It panics on invalid paths. For that reason, it should not be used with user-provided paths.
```
Comment (rumyantseva) [inline internal/types/path.go:75]: Can we explain what paths we consider invalid?

**Review by rumyantseva — COMMENTED** [review]
Added a doc suggestion.

Please check the linter (it complains).

File: internal/types/path.go (hunk 1 above)
Comment (AlekSi) [reply internal/types/path.go:75]: Sure, in #1954

**Review by rumyantseva — APPROVED** [review]
LGTM as TODO for the actual task

**Review by noisersup — APPROVED** [review]
:shipit:

**Pull request merged**